<?xml version="1.0"?>

<launch>
  <!-- Bluerov robot transforms -->
  <include file="$(find bluerov)/launch/description.launch" />

  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" output="screen" respawn="true">
      <param name="num_worker_threads" value="2"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="fake_video_stream" output="screen" args="load tag_response_test/FakeVideoStreamNodelet nodelet_manager">
    <param name="publish_rate" type="int" value="30"/>
    <param name="fake_video_image_path" type="string" value="$(find tag_response_test)/images/test_image.png"/>
    <rosparam command="load" file="$(find bluerov_controller)/param/camera_calibration_720_540_barbados_ocean.yaml"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="stag_ros" output="screen" args="load stag_ros/StagNodelet nodelet_manager" respawn="true">
    <param name="camera_image_topic" type="string" value="/fake_video_stream/fake_video_stream/image_raw" />
    <param name="camera_info_topic" type="string" value="/fake_video_stream/fake_video_stream/camera_info" />
    <param name="tag_id_type" type="int" value="21" />
    <param name="marker_frame_prefix" type="string" value="/ar_marker_" />
    <param name="default_marker_size" type="double" value="0.166" />
    <param name="output_frame_id"     type="string" value="/cam_0_optical_frame" />
    <param name="image_frame_id"      type="string" value="/cam_0_optical_frame" />
    <param name="marker_message_topic" type="string" value="/bluerov_controller/ar_tag_detector" />
    <param name="use_marker_bundles" type="bool" value="true" />
    <rosparam command="load" param="marker_sizes_by_id" file="$(find bluerov_controller)/param/marker_sizes_by_id.yaml"/>
    <rosparam command="load" file="$(find bluerov_controller)/param/marker_bundles.yaml"/>
  </node>

  <node pkg="tag_response_test" type="marker_subscriber.py" name="marker_subscriber" output="screen">
      <param name="ar_tag_topic" type="string" value="/bluerov_controller/ar_tag_detector"/>
      <param name="output_file" type="string" value="$(find tag_response_test)/output_data/30_fps-one-tag-near:.csv"/>
      <param name="number_samples" type="int" value="1000"/>
  </node>
</launch>
